### ISSUE TITLE
CI methodology

### ISSUE THREAD

**Issue by nikohansen** [numbbo/coco#1641]
It turns out that our master branch has gone broke but we did not know this until I manually triggered the tests more or less out of curiosity. This suggests a problem with our testing methodology.

**Comment by nikohansen** [comment]
Currently the only smooth way to change the master branch is a pull request _from the development branch_. This prevents quick bug fixes to happen and suggests a problem with our testing logic.

**Comment by dtusar** [comment]
The master build fails because the test tries to download the algorithm from `http://coco.gforge.inria.fr/data-archive/2009/BFGS_ros_noiseless.tgz`. But the file is not there because the repository structure has changed.

Regarding the testing methodology, if we schedule to test the master branch once in a while (maybe once a week) then we would probably catch this kind of issues.

**Comment by nikohansen** [comment]
Bringing back `http://coco.gforge.inria.fr/data-archive/2009/BFGS_ros_noiseless.tgz` seems not to fix the test of the master branch.

**Comment by dtusar** [comment]
Now the non-existing file/location is `http://coco.gforge.inria.fr/data-archive/bbob-biobj/2016/RS-4.tgz`.

**Comment by brockho** [comment]
I added the `RS-4.tgz` again to the required location. Which test shall I rerun to see whether the problem remains?

**Comment by dtusar** [comment]
Start with the `Coco-master-linux`.

**Comment by brockho** [comment]
`Coco-master-linux` is back green :-) I re-started the other tests on the `master` branch as well now.

**Comment by dtusar** [comment]
Even before you added the required files back to the correct location some incomplete `tgz` files were generated on the slaves. So, I had to delete them manually. In the development branch we changed the location to which the files are downloaded, so we shouldn't worry about this any more. 

On the windows slave the test finished before I deleted the files. That's why the test failed. I run it again and now it should pass.
